src/Package/TestApp/Utility/ViewTreeDump.cs

```diff
@@ -36,27 +35,36 @@ internal class ViewTreeDump {
         }
 
         private static void CompareVisualTree(VisualTreeObject actual, VisualTreeObject expected, bool compareProperty = true) {
+            bool visible = true;
+            
             // compare
-            actual.Name.ShouldBeEquivalentTo(expected.Name);
+            actual.Name.Should().BeEquivalentTo(expected.Name);
 
-            if (compareProperty) {
+            var visibility = actual.Properties.FirstOrDefault(p => p.Name == "Visibility");
+            if (visibility != null) {
+                visible = (visibility.Value != "Collapsed");
+            }
+
+            if (compareProperty && visible) {
                 var filteredActual = actual.Properties.Where(p => SupportedWpfProperties.IsSupported(p.Name));
                 var filteredExpected = expected.Properties.Where(p => SupportedWpfProperties.IsSupported(p.Name));
```

Comment:
This whole thing can be replaced with `filteredActual.Should().BeEquivalentTo(filteredExpected)`